### PR TITLE
Change lexical scope tuning

### PR DESCRIPTION
**Pull request by haslinghuis**

### CODE REVIEW

File: src/js/tabs/pid_tuning.js [hunk 1]
```diff
@@ -1326,9 +1326,9 @@ TABS.pid_tuning.initialize = function (callback) {
 
         // Add a name to each row of PIDs if empty
         $('.pid_tuning tr').each(function(){
-            for(i = 0; i < FC.PID_NAMES.length; i++) {
+            for (const i in FC.PID_NAMES) {
```
Comment (mikeller) [inline src/js/tabs/pid_tuning.js:1329]: `for ( of )` or leave as is.

File: src/js/tabs/pid_tuning.js [hunk 2]
```diff
@@ -1978,12 +1978,12 @@ TABS.pid_tuning.initialize = function (callback) {
...
             .then(function () {
-                var promise;
                 if (semver.gte(FC.CONFIG.apiVersion, CONFIGURATOR.API_VERSION_MIN_SUPPORTED_PID_CONTROLLER_CHANGE) && semver.lt(FC.CONFIG.apiVersion, API_VERSION_1_31)) {
                     FC.PID.controller = pidController_e.val();
-                    promise = MSP.promise(MSPCodes.MSP_SET_PID_CONTROLLER, mspHelper.crunch(MSPCodes.MSP_SET_PID_CONTROLLER));
+                    MSP.promise(MSPCodes.MSP_SET_PID_CONTROLLER, mspHelper.crunch(MSPCodes.MSP_SET_PID_CONTROLLER));
+                } else {
+                    console.log(`Unsupported API version: ${FC.CONFIG.apiVersion}`);
                 }
-                return promise;
```
Comment (mikeller) [inline src/js/tabs/pid_tuning.js:1986]: Why was this removed?

File: src/js/tabs/pid_tuning.js [hunk 3]
```diff
@@ -2164,8 +2188,8 @@ TABS.pid_tuning.checkRC = function() {
     if (!this.oldRC) { this.oldRC = [FC.RC.channels[0], FC.RC.channels[1], FC.RC.channels[2]]; }
 
     // Monitor FC.RC.channels and detect change of value;
-    var rateCurveUpdateRequired = false;
-    for(var i=0; i<this.oldRC.length; i++) { // has the value changed ?
+    let rateCurveUpdateRequired = false;
+    for (let i=0; i<this.oldRC.length; i++) { // has the value changed ?
```
Comment (mikeller) [inline src/js/tabs/pid_tuning.js:2192]: Whitespace. :space_invader:

File: src/js/tabs/pid_tuning.js [hunk 4]
```diff
@@ -2253,7 +2277,7 @@ TABS.pid_tuning.updateRatesLabels = function() {
             y -= (height/2); if(y<0) y=0; else if(y>context.height) y=context.height; // prevent balloon from going out of canvas
 
             // check that the balloon does not already overlap
-            for(var i=0; i<dirty.length; i++) {
+            for(const i in dirty) {
```
Comment (mikeller) [inline src/js/tabs/pid_tuning.js:2280]: `for ( of )` or leave as is.

File: src/js/tabs/pid_tuning.js [hunk 5]
```diff
@@ -2388,33 +2411,35 @@ TABS.pid_tuning.updateRatesLabels = function() {
                 );
             }
             // then display them on the chart
-            for(var i=0; i<balloons.length; i++) balloons[i].balloon();
+            for(const i in balloons) {
```
Comment (mikeller) [inline src/js/tabs/pid_tuning.js:2414]: `for ( of )` or leave as is.

**Review by mikeller — CHANGES_REQUESTED** [review]

File: src/js/tabs/pid_tuning.js (hunk 3 above)
Comment (haslinghuis) [reply src/js/tabs/pid_tuning.js:2192]: 👾

**Comment by haslinghuis** [comment]
Fixed as requested per review. Added fixing spaces in several places to be more consistent.
